src/shared/systemUtilities.ts

```diff
@@ -38,4 +40,29 @@ export class SystemUtilities {
             })
         })
     }
+
+    /**
+     * Searches for `tsc` in the current workspace, or the system (tries `tsc`
+     * using current $PATH).
+     *
+     * @returns fullpath if found in the workspace, "tsc" if found in current $PATH, else undefined.
+     */
+    public static async findTypescriptCompiler(): Promise<string | undefined> {
+        const foundUris = await vscode.workspace.findFiles('**/node_modules/.bin/{tsc,tsc.cmd}', undefined, 1)
+        const tscPaths = []
+        if (foundUris.length > 0) {
+            tscPaths.push(foundUris[0].fsPath)
+        }
+        tscPaths.push('tsc') // Try this last.
+
+        for (const tsc of tscPaths) {
+            // Try to run "tsc -v".
```

Comment:
Let me confirm if this works in Windows?

Reply:
OK, looks like it works:
* Code-invoke plain js app works without crashing
* Code-invoke TS app fails if `npm install` hasn't been run previously
* Code-invoke TS app succeeds if `npm install` is run